client/src/Pages/ItineraryDetails.jsx

```diff
@@ -0,0 +1,184 @@
+                console.error('verifyCookie error:', error);
+                navigate("/login");
+            }
+        };
+
+        const getUser = async () => {
+            if (userId === "" && email !== "") {
+                try {
+                    if (data.data) {
+                        setUserId(await data.data.find(user => user.email === email)._id);
+                    }
+                } catch (error) {
+                    console.error('getUser error:', error);
+                }
+            }
+        }
+
+        const getItinerary = async () => {
+            if (userId !== "") {
+                try {
+                    const res = await axios.get(`http://localhost:4000/itineraries/${userId}`, { withCredentials: true });
+                    const selectedItinerary = res.data.data.find(itinerary => itinerary._id === id);
+                    console.log(selectedItinerary);
+                    if (selectedItinerary) {
+                        setItineraryId(selectedItinerary._id);
+                        setItineraryTitle(selectedItinerary.title);
+                    } else {
+                        console.error('Selected itinerary not found');
+                    }
+                } catch (error) {
+                    console.error('getItinerary error:', error);
+                }
+            }
+        }
+          
+        verifyCookie();
+        getUser();
+        getItinerary();
+
+    }, [navigate, cookies, removeCookie, email, data, loading, userId, id, itineraryId]);
+
+    useEffect(() => {
+        if (loading && userId && itineraryId !== "") {
+            axios.get(`http://localhost:4000/activities/${itineraryId}`, { withCredentials: true })
+                .then((response) => {
+                    setActivities(response.data.data);
+                    setLoading(false);
+                })
+                .catch((error) => {
+                    console.log(error);
+                    setLoading(true);
+                    console.log("error");
+                });
+        }
+        console.log(activities);
+    }, [userId, itineraryId, activities, loading]);
+
+    const handleDeleteActivity = (activity) => {
+        if (window.confirm('Are you sure you wish to delete this item?')) {
+           deleteActivity(activity) 
+       } 
+   }
+
+   const deleteActivity = async (activity) => {
+       try {
+           const res = await axios.delete(`http://localhost:4000/activities/${itineraryId}/${activity._id}`, { withCredentials: true });
+           if (res.data.success) {
+               window.location.reload();
+           } else {
+               handleError("Error, activity unable to be deleted!");
+           }
+       } catch (error) {
+           console.log(error);
+       }
+   }
+
+   const handleError = (err) => {
+       toast.error(err, {
+           position: "bottom-left",
+       });
+   }
+
+   const sortActivitiesByDays = (activities) => {
```

Comment:
i think there might be a bug somewehere here since i tried adding two activities that are two days apart (one on 8 jan and the other on 10 jan) but they appear under Day 1 and then Day 2 instead of Day 1 then Day 3

![image](https://github.com/chewyxx/WinTech/assets/117144224/729fb24f-b6e3-4384-8ce7-9a119a77e5ab)
![image](https://github.com/chewyxx/WinTech/assets/117144224/d97d98d2-f6c9-49ee-8e9e-bcc73a22decb)